cli/flox-rust-sdk/src/providers/git.rs

```diff
@@ -638,6 +641,28 @@ impl From<GitCommandError> for GitRemoteCommandError {
                 debug!("Ref not found: {ref_name}");
                 GitRemoteCommandError::RefNotFound(ref_name.to_string())
             },
+            GitCommandError::BadExit(_, _, ref stderr)
+                if stderr.starts_with(REMOTE_BRANCH_NOT_FOUND_ERR_PREFIX) =>
+            {
+                let branch_name = stderr
+                    .strip_prefix(REMOTE_BRANCH_NOT_FOUND_ERR_PREFIX)
+                    .unwrap();
+                let branch_name = branch_name.strip_suffix(" to clone").unwrap_or(branch_name);
+                debug!("Could not find remote branch: {branch_name}");
+                GitRemoteCommandError::RefNotFound(branch_name.to_string())
+            },
+            GitCommandError::BadExit(_, _, ref stderr)
+                if stderr.starts_with(REMOTE_BRANCH_NOT_FOUND_IN_UPSTREAM_ERR_PREFIX) =>
+            {
+                let branch_name = stderr
+                    .strip_prefix(REMOTE_BRANCH_NOT_FOUND_IN_UPSTREAM_ERR_PREFIX)
+                    .unwrap();
+                let branch_name = branch_name
+                    .strip_suffix(" not found in upstream origin")
+                    .unwrap_or(branch_name);
+                debug!("Could not find remote branch in upstream: {branch_name}");
+                GitRemoteCommandError::RefNotFound(branch_name.to_string())
+            },
```

Comment:
I agree, this doesn't seem to add value, esp. since we can trace the branch name in the verbose logs anyway. 
I would love to do a follow up PR to remove the branch name from the error.